administration/src/cards/extensions/__tests__/KoblenzReferenceNumberExtension.test.tsx

```diff
@@ -0,0 +1,157 @@
+      fireEvent.change(input, { target: { value: '12345' } })
+      expect(mockSetValue).toHaveBeenCalledWith({ koblenzReferenceNumber: '12345' })
+    })
+
+    it('should not show error message if field was not left and required is false', () => {
+      const initialValue = '123@45'
+      const { queryByText } = renderWithTranslation(
+        <KoblenzReferenceNumberExtension.Component
+          value={{ koblenzReferenceNumber: initialValue }}
+          setValue={mockSetValue}
+          isValid={false}
+          showRequired={false}
+        />
+      )
+      expect(queryByText('Das Aktenzeichen enthält ungültige Sonderzeichen.')).toBeNull()
+    })
+
+    it('should show error for special characters', () => {
+      const initialValue = '123@45'
+      const { getByText, getByDisplayValue } = renderWithTranslation(
+        <KoblenzReferenceNumberExtension.Component
+          value={{ koblenzReferenceNumber: initialValue }}
+          setValue={mockSetValue}
+          isValid={false}
+          showRequired
+        />
+      )
+      const input = getByDisplayValue(initialValue)
+      fireEvent.blur(input)
+      expect(getByText('Das Aktenzeichen enthält ungültige Sonderzeichen.')).toBeTruthy()
+    })
+
+    it('should show error if the value is too short', () => {
+      const initialValue = '123'
+      const { getByDisplayValue, getByText } = renderWithTranslation(
+        <KoblenzReferenceNumberExtension.Component
+          value={{ koblenzReferenceNumber: initialValue }}
+          setValue={mockSetValue}
+          isValid={false}
+          showRequired
+        />
+      )
+      const input = getByDisplayValue(initialValue)
+      fireEvent.blur(input)
+      expect(getByText('Das Aktenzeichen muss eine Länge zwischen 4 und 15 haben.')).toBeTruthy()
+    })
+
+    it('should show error if the value exceeds the max length', () => {
+      const initialValue = '123KKdas22sdas23'
+      const { getByDisplayValue, getByText } = renderWithTranslation(
+        <KoblenzReferenceNumberExtension.Component
+          value={{ koblenzReferenceNumber: initialValue }}
+          setValue={mockSetValue}
+          isValid={false}
+          showRequired
+        />
+      )
+      const input = getByDisplayValue(initialValue)
+      fireEvent.blur(input)
+      expect(getByText('Das Aktenzeichen muss eine Länge zwischen 4 und 15 haben.')).toBeTruthy()
+    })
+
+    it('should clear input when clear button is clicked', () => {
+      const { getByRole } = renderWithTranslation(
+        <KoblenzReferenceNumberExtension.Component
+          value={{ koblenzReferenceNumber: '12345' }}
+          setValue={mockSetValue}
+          isValid
+          showRequired={false}
+        />
+      )
+
+      const clearButton = getByRole('button')
+      fireEvent.click(clearButton)
+      expect(mockSetValue).toHaveBeenCalledWith({ koblenzReferenceNumber: '' })
+    })
+
+    describe('isValid', () => {
```

Comment:
can be merged with component tests (same reason as discussed in birthday extension)